pkg/image/image.go

```diff
@@ -392,26 +391,38 @@ func ChangeImageDomain(imageString string, newDomain string) (types.ImageReferen
 	return newImage, nil
 }
 
-func WithoutTag(image string) (string, error) {
-	fullImage, err := AddDefaultRegistry(image, "place.holder.com")
-	if err != nil {
-		return image, err
+// SplitImage Returns an image tag if it exists, an image digest if it exists, the image without the tag and digest, and any errors; in this order.
+// images with both a tag and digest are not supported
+func SplitImage(imageArg string) (string, string, string, error) {
+	userImage := imageArg
+	//Parse the image using the docker transport
+	transportName := alltransports.TransportFromImageName(userImage)
+	if transportName == nil {
+		userImage = "docker://" + userImage
 	}
-	named, err := reference.ParseNamed(fullImage)
+	imageRef, err := alltransports.ParseImageName(userImage)
 	if err != nil {
-		return image, err
-	}
-	named = reference.TagNameOnly(named)
-	tagged, ok := named.(reference.NamedTagged)
-	if !ok {
-		tmp := fmt.Sprintf("error removing image tag from string %s", image)
-		return image, errors.New(tmp)
-	}
-	justImage, found := strings.CutSuffix(named.String(), ":"+tagged.Tag())
-	if found {
-		return justImage, nil
-	}
+		return "", "", "", fmt.Errorf("error parsing image %s with error %v", imageArg, err)
+	}
+	if imageRef.DockerReference() == nil {
+		return "", "", "", fmt.Errorf("error parsing image %s not a valid docker reference", imageArg)
```

Comment:
"error parsing image, %s is not a valid docker reference"

Reply:
Ok! Thanks!